### ISSUE TITLE
Notifications: sending emails failed

### ISSUE THREAD

**Issue by jframeau** [Graylog2/graylog2-server#12900]
Hi,

Just upgraded Graylog 4.2 to 4.3. Notifications was ok running 4.2.
Now, I get an exception and this message on web console when testing:

Error: Notification has email recipients and is triggered, but sending emails failed. null

In overview:

 An error occurred while trying to send an email! (triggered 18 minutes ago)
The Graylog server encountered an error while trying to send an email. This is the detailed error message: java.lang.NullPointerException

Exception:

```
org.graylog.scheduler.JobExecutionException: Failed permanently to execute notification, giving up - <627e625ff333e2502d11e31c/Notification_Mail/email-notification-v1>
        at org.graylog.events.notifications.EventNotificationExecutionJob.execute(EventNotificationExecutionJob.java:156) ~[graylog.jar:?]
        at org.graylog.scheduler.JobExecutionEngine.executeJob(JobExecutionEngine.java:172) ~[graylog.jar:?]
        at org.graylog.scheduler.JobExecutionEngine.lambda$handleTrigger$2(JobExecutionEngine.java:150) ~[graylog.jar:?]
        at com.codahale.metrics.Timer.time(Timer.java:151) ~[graylog.jar:?]
        at org.graylog.scheduler.JobExecutionEngine.handleTrigger(JobExecutionEngine.java:150) ~[graylog.jar:?]
        at org.graylog.scheduler.JobExecutionEngine.lambda$execute$0(JobExecutionEngine.java:119) ~[graylog.jar:?]
        at org.graylog.scheduler.worker.JobWorkerPool.lambda$execute$0(JobWorkerPool.java:121) ~[graylog.jar:?]
        at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:180) [graylog.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_332]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_332]
        at com.codahale.metrics.InstrumentedThreadFactory$InstrumentedRunnable.run(InstrumentedThreadFactory.java:66) [graylog.jar:?]
        at java.lang.Thread.run(Thread.java:750) [?:1.8.0_332]
Caused by: org.graylog.events.notifications.PermanentEventNotificationException: Notification has email recipients and is triggered, but sending emails failed. null
        at org.graylog.events.notifications.types.EmailEventNotification.execute(EmailEventNotification.java:93) ~[graylog.jar:?]
        at org.graylog.events.notifications.EventNotificationExecutionJob.execute(EventNotificationExecutionJob.java:135) ~[graylog.jar:?]
```

* Graylog Version: graylog-server-4.3.1-1 / raylog-integrations-plugins-4.3.1-1
* Java Version: openjdk version "1.8.0_332" 
* Elasticsearch Version: elasticsearch-oss-7.10.2-1
* MongoDB Version: mongodb-org-server-4.2.20-1
* Operating System: Rocky Linux release 8.6 (Green Obsidian)
* Browser version: Firefox 101.0.1

Email configuration:

```
server.conf:transport_email_enabled = true
server.conf:transport_email_hostname = 127.0.0.1
server.conf:transport_email_port = 25
server.conf:transport_email_use_auth = false
server.conf:transport_email_use_tls = false
server.conf:transport_email_use_ssl = false
```

Thx

**Comment by jframeau** [comment]
New log:

The Graylog server encountered an error while trying to send an email. This is the detailed error message: java.lang.NullPointerException: Cannot invoke "String.length()" because "s" is null

**Comment by HungryHowies** [comment]
@jframeau 
Was this your post?

https://community.graylog.org/t/graylog-4-3-2-failed-to-send-email-notification/24462

**Comment by jframeau** [comment]
@HungryHowies no, but thank you very much for the hint! 

Notifications are back after fixing transport_email_from_email (was commented). 

But not sure to undestand the difference with Sender in Graylog UI.

thx

**Comment by filviu** [comment]
To add some notes:

I was running without having defined `transport_email_from_email` (actually `GRAYLOG_TRANSPORT_EMAIL_FROM_EMAIL` as I use docker) up to the latest 4.2.x

Once upgraded to 4.3.2 I couldn't send notifications anymore even if **Sender (Optional)** is defined in the GUI

P.S. The error message is really unhelpful. The upgrade notes make no mention of `transport_email_from_email` becoming mandatory. That made for a really bad upgrade experience.

**Comment by mpfz0r** [comment]
Thanks for the reports, this regression will be fixed with the upcoming 4.3.4 bugfix release.
